mlflow/entities/_mlflow_object.py

```diff
@@ -20,3 +21,37 @@ def from_proto(cls, proto):
     @classmethod
     def from_dictionary(cls, the_dict):
         return cls(**the_dict)
+
+    def __repr__(self):
+        return serialize(self)
+
+
+def serialize(obj):
+    return _MLflowObjectPrinter().serialize(obj)
+
+
+def get_classname(obj):
+    return obj.__module__ + "." + obj.__class__.__qualname__
+
+
+class _MLflowObjectPrinter(object):
+    _MAX_LIST_LEN = 2
+
+    def __init__(self):
+        super(_MLflowObjectPrinter, self).__init__()
+        self.printer = pprint.PrettyPrinter()
+
+    def serialize(self, obj):
+        if isinstance(obj, _MLflowObject):
+            return "<%s: %s>" % (get_classname(obj), self.serialize_entity(obj))
```

Comment:
I'd include just the last part of the class name: Experiment instead of mlflow.entities.experiment.Experiment.